src/toolchains.rs

```diff
@@ -322,6 +312,32 @@ impl Toolchain {
         output
     }
 
+    fn set_cargo_args_and_envs(&self, cmd: &mut Command, cfg: &Config) {
+        let rustup_name = format!("+{}", self.rustup_name());
+        cmd.arg(&rustup_name);
+        if cfg.args.command_args.is_empty() {
+            cmd.arg("build");
+        } else {
+            cmd.args(&cfg.args.command_args);
+        }
+        if cfg.args.pretend_to_be_stable && self.is_current_nightly() {
+            // Forbid using features
+            cmd.env(
+                "RUSTFLAGS",
+                format!(
+                    "{} -Zenable-features=",
+                    std::env::var("RUSTFLAGS").unwrap_or_default()
+                ),
+            );
+            // Make rustc report a stable version string derived from the current nightly's version string.
+            let version = rustc_version::version_meta().unwrap().semver;
+            cmd.env(
+                "RUSTC_FORCE_RUSTC_VERSION",
+                format!("{}.{}.{}", version.major, version.minor, version.patch),
```

Comment:
This doesn't actually change the version string, does it? My understanding is that it only changes the stable crate id and incremental cache version. I'm not sure what benefit that brings.

Reply:
I tested it on the rustc side, and it works: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=0bf99dd363de2b1659c24e9e357a34e7

Reply:
hmm looks like I screwed up testing